src/server/nonClassEvent/nonClassEvent.controller.ts

```diff
@@ -49,4 +79,37 @@ export class NonClassEventController {
       );
     return nonClassEvents;
   }
+
+  @Post('/')
+  @ApiOperation({
+    summary: 'Create a new non-class parent',
+    description: 'Creates a new non-class parent and populates all related non-class events (one assocaited with each semester in the database)',
+  })
+  @ApiCreatedResponse({
+    type: NonClassParentResponse,
+    description: 'The non-class parent was created along with one non-class event for each semester in the database',
+  })
+  @ApiBody({
+    type: CreateNonClassParentDTO,
+    required: true,
+  })
+  public async create(@Body() { area, ...parent }: CreateNonClassParentDTO):
+  Promise<NonClassParentResponse> {
+    try {
+      const dbArea = await this.areaRepository.findOneOrFail({
+        where: {
+          name: area,
+        },
+      });
```

Comment:
The `CreateNonClassParentDTO` says that the `area` field must be a `UUID`, so I expect that this is always going to fail.

Reply:
Ah, yes - good catch, I've updated this to search using `area` as the PK id(which is what it was supposed to be in the first place)